### PR TITLE
Tunables service fix and theme changes

### LINKED ISSUE #618: [Request]: Notes Text Field in Player Database
### Problem

_No response_

### Solution

Would be nice to have a note text field in the player database so you can note down toxic behavior or something else.

### Reason

/

### Additional context

_No response_

### LINKED ISSUE #1433: [Bug]: Operations Center unavailable with menu running
### Describe the bug

"Operations Center is unavailable right now" when approaching the screen with menu running. Unloading menu and switching sessions fixes this.

### Steps To Reproduce

1. Load menu whenever
2. Go to Avenger
3. Enjoy the view

### Expected Behavior

Works smoothly

### Store

Steam

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### LINKED ISSUE #1442: [Bug]: [FATAL/tunables_service.cpp:87] Failed to cache tunables pops up after injecting into game
### Describe the bug

the mod menu worked fine but I closed it and pressed unload and ever since then it just says [FATAL/tunables_service.cpp:87] Failed to cache tunables and stays on the tab that shows fps and players in your lobby I have uninstalled and reinstalled both the mod menu and the xenos injector nothing works

### Steps To Reproduce

i loaded gta up
injected the mod menu 
pressed unload on the mod menu 
restarted my game
loaded it back up 
now it gets to [FATAL/tunables_service.cpp:87] Failed to cache tunables and stops after [INFO/tunables.hpp:25] Snow loaded

### Expected Behavior

idk

### Screenshots and/or Logs

```
<paste your logs here>
```
Screenshot 2023-06-16 154207
Screenshot 2023-06-16 153838

### Store

Steam

### Additional context

im really not to sure whats going on and i need some help

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### LINKED ISSUE #1469: [Bug]: Interiors Vehicle Modification Shop doesnt work when menu is loaded
### Describe the bug

When menu is loaded Agency, AutoShop and Office vehicle mod shop not work at all. Nothing happen when tryin pressing button to enter modshop. Motorcycles club Vehicle ModShop is works but when key E is pressed vehicle acts like its outside (peform Jump or Boost).

### Steps To Reproduce

1. Load Menu
2. Go to Agency, Office or AutoShop
3. Try to enter Vehicle Modification by pressin Q (AutoShop) or E for Agency or Office
4. Nothing happen.

### Expected Behavior

All interiors vehicle modification functions should work fine

### Screenshots and/or Logs

_No response_

### Store

Epic Games

### Additional context

Tried to disable all Vehicle options but it not help to fix the issue. Build is latest but that happen for previous builds too.

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### LINKED ISSUE #1498: [Bug]: Can't use the operation terminal in the avanger when Yimmenu loaded (mercenaries update)
### Describe the bug

The GTA Online San Andreas Mercenaries introduced a new Operations Terminal for the Avenger. If YimMenu is loaded, I can't use it, because it says "The Operation Terminal ist not available" (or something like that, translated from german).

### Steps To Reproduce

1. Start GTA online
2. Load Yimmenu
3. Enter the Avenger and go to the Operations Terminal

### Expected Behavior

The Operations Terminal should be usable.

### Screenshots and/or Logs

```
<paste your logs here>
```

### Store

Steam

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### LINKED ISSUE #1504: [Bug]: Unable to use the operator terminal and Avenger's autopilot mode
### Describe the bug

Unable to use the operator terminal and Avenger's autopilot mode

### Steps To Reproduce

Unable to use the operator terminal and Avenger's autopilot mode

### Expected Behavior

Unable to use the operator terminal and Avenger's autopilot mode

### Screenshots and/or Logs

```
<paste your logs here>
```

### Store

Steam

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### PR DESCRIPTION
**Pull request by maybegreat48**
* Added a script connection service (might be buggy, haven't tested it completely)
* Added an option to display the local player's position in the overlay
* Added notes to the player database
* Changed the default color palette from black to green (and added some basic theme customization)

Fixes #1504 
Fixes #1498
Fixes #1469 
Fixes #1442 
Fixes #1433 
Closes #618

### CODE REVIEW

**Review by Yimura — DISMISSED** [review]

File: src/services/tunables/tunables_service.cpp [hunk 1]
```diff
@@ -60,9 +60,9 @@ namespace big
 
 				if (SCRIPT::HAS_SCRIPT_WITH_NAME_HASH_LOADED(RAGE_JOAAT("tuneables_processing")))
 				{
-					std::uint64_t args[] = {6, 27, 1}; // TODO: check args
+					std::uint64_t args[] = {6, 27}; // TODO: check args
 
-					int id = SYSTEM::START_NEW_SCRIPT_WITH_NAME_HASH_AND_ARGS(RAGE_JOAAT("tuneables_processing"), (Any*)args, sizeof(args) / 8, 5050);
+					int id = SYSTEM::START_NEW_SCRIPT_WITH_NAME_HASH_AND_ARGS(RAGE_JOAAT("tuneables_processing"), (Any*)args, sizeof(args) / 8, 1424);
```
Comment (xiaoxiao921) [inline src/services/tunables/tunables_service.cpp]: you defined a constexpr for this, may aswell use it

File: src/views/core/view_overlay.cpp [hunk 2]
```diff
@@ -26,6 +28,16 @@ namespace big
 				ImGui::Text(std::format("Players: {}/{}", network_player_mgr->m_player_count, network_player_mgr->m_player_limit)
 				                .c_str());
 
+			if (g.window.ingame_overlay.show_position && g_local_player)
+			{
+				ImGui::Separator();
+
+				ImGui::Text("Pos: %.2f, %.2f, %.2f",
+				    g_local_player->get_position()->x,
```
Comment (xiaoxiao921) [inline src/views/core/view_overlay.cpp]: a local variable for this would be nice

File: src/services/script_connection/script_connection.cpp [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+
+		if (m_startup_done && m_target && !m_target->get()->is_valid())
+		{
+			if (m_debug_logging)
+				LOG(VERBOSE) << "Target has left, cleaning up";
+
+			return true;
+		}
+
+#if 0
```
Comment (xiaoxiao921) [inline src/services/script_connection/script_connection.cpp:372]: why is this here

File: src/gui.cpp [hunk 4]
```diff
@@ -128,24 +128,62 @@ namespace big
...
+		ImGui::PushStyleColor(ImGuiCol_ButtonHovered, button_color);
+		ImGui::PushStyleColor(ImGuiCol_ButtonActive, button_active_color);
+		ImGui::PushStyleColor(ImGuiCol_FrameBg, frame_color);
+		ImGui::PushStyleColor(ImGuiCol_FrameBgHovered, frame_hovered_color);
+		ImGui::PushStyleColor(ImGuiCol_FrameBgActive, frame_active_color);
+	}
+
+	void gui::pop_theme_colors()
+	{
+		ImGui::PopStyleColor();
```
Comment (xiaoxiao921) [inline src/gui.cpp]: this whole part scream for a wrapper function that also increment a variable which is then used for forloop

File: src/invoker.cpp [hunk 5]
```diff
@@ -43,14 +43,16 @@ namespace big
 			[this, hash, handler] {
 				__try
 				{
-					_call_asm(&m_call_context, handler, g_pointers->m_gta.m_native_return);
-					// handler(&m_call_context);
+					//_call_asm(&m_call_context, handler, g_pointers->m_gta.m_native_return);
```
Comment (xiaoxiao921) [inline src/invoker.cpp]: remove commented code completly
could also remove the .asm file I guess?

**Review by xiaoxiao921 — DISMISSED** [review]
small nitpicks here and there, good amount of work nontheless

File: src/hooks/script/script_vm.cpp [hunk 6]
```diff
@@ -56,6 +56,12 @@ namespace big
 
 	rage::eThreadState hooks::script_vm(uint64_t* start_stack, uint64_t** scr_globals, rage::scrProgram* program, rage::scrThreadContext* ctx)
 	{
+		if (program->m_name_hash == RAGE_JOAAT("net_tunable_check")) // not sure what starts this, but it tries to crash the game and must be blocked
```
Comment (SkiddyToast) [inline src/hooks/script/script_vm.cpp]: This script has never been used/started, there's no point in adding this.

File: src/services/script_connection/script_connection.cpp (hunk 3 above)
Comment (maybegreat48) [reply src/services/script_connection/script_connection.cpp:372]: It is currently broken and causes the script to unload early. Keeping the code as it wouldn't be stored in the history when the PR gets squashed and merged as a single commit

File: src/backend/commands/player/toxic/kick_from_interior.cpp [hunk 7]
```diff
@@ -26,107 +27,19 @@ namespace big
...
+				auto cxn = g_script_connection_service->create_connection("am_mp_property_int", player);
+				cxn->set_host_broadcast_size(1359);
+				cxn->set_player_broadcast_size(1185);
+				cxn->set_no_remote_start(true);
+				cxn->set_wait_for_host_broadcast_data(true);
+				cxn->set_quick_cleanup(true);
+				cxn->set_debug_logging(true);
+				cxn->set_broadcast_modify_callback([](rage::scrThread* thread, uint64_t* server_vars, uint64_t* client_vars) {
+					server_vars[718] = 1;
+				});
```
Comment (frostyperflix) [inline src/backend/commands/player/toxic/kick_from_interior.cpp:39]: This looks really clean. 😄
Very nice addition!

**Review by aoqia194 — COMMENTED** [review]
Looks pretty good to me (apart from what people have already commented about). I am not a code-reviewer and I just quickly looked through so my review is just a comment as it provides no significance. I am curiously waiting for this to be merged as I sense it might fix a couple issues that aren't related.

**Review by Yimura — APPROVED** [review]

**Pull request merged**